exchange/exchange.go

```diff
@@ -102,9 +107,22 @@ func (e *exchange) HoldAuction(ctx context.Context, bidRequest *openrtb.BidReque
 		e.me.RecordImps(impLabels)
 	}
 
+	// Make our best guess if GDPR applies
+	usersyncIfAmbiguous := e.UsersyncIfAmbiguous
+	var geo *openrtb.Geo = nil
+	if bidRequest.User != nil && bidRequest.User.Geo != nil {
+		geo = bidRequest.User.Geo
+	} else if bidRequest.Device != nil && bidRequest.Device.Geo != nil {
+		geo = bidRequest.Device.Geo
+	}
+	if geo != nil {
+		if _, found := e.gdprCountries[geo.Country]; found {
```

Comment:
I looks like `UsersyncIfAmbiguous` must be enabled for the Geo check to matter, otherwise it's always false. Should it instead be "if you're an EU country, enforce GDPR... otherwise fall back to UsersyncIfAmbiguous". This would better handle the case of a EU request somehow being routed to a server outside of the EU.

Reply:
It is confusing, `UsersyncIfAmbiguous` is the opposite of enforcing the GDPR by default. If if is false, we always assume gdpr applies if no signal. So if we always assume gdpr, then it doesn't make sense to check if you are in an eu country. I should put this in an if statement to save a few cycles if we are always assuming gdpr is active.